server/src/main/java/org/opensearch/search/startree/StarTreeTraversalUtil.java

```diff
@@ -169,12 +167,15 @@ private static StarTreeResult traverseStarTree(StarTreeValues starTreeValues, Ma
             }
 
             if (remainingPredicateColumns.contains(childDimension)) {
-                long queryValue = queryMap.get(childDimension); // Get the query value directly from the map
-                StarTreeNode matchingChild = starTreeNode.getChildForDimensionValue(queryValue);
-                if (matchingChild != null) {
-                    queue.add(matchingChild);
-                    foundLeafNode |= matchingChild.isLeaf();
+                List<DimensionFilter> dimensionFilters = starTreeFilter.getFiltersForDimension(childDimension);
+                final boolean[] tempFoundLeafNodes = new boolean[1];
```

Comment:
why not just a boolean ?

Reply:
Can't update inside anonymous class. 
Becomes a Compilation error of `Effective final variable must be used`